### ISSUE TITLE
[FIREFOX BUG?] FF remembers windows with GH sidebar or not only if window where you close FF has not.

### ISSUE THREAD

**Issue by user0022** [madprops/grasshopper#102]
Hello!
I use Firefox ESR 115, under Lubuntu 22.04 OS.

*

In short, if you prefer to have only some windows containing a GH sidebar, and Firefox remembering them at each restart, visibly the window from which you close FF must contain no sidebar, otherwise next time it will open a sidebar in every window.

I post this thing here even if it is a Firefox bug (if I understand what I see). So GH probably cannot fix it (but maybe it can override this FF behaviour, as a non-developper my idea about it may be wrong). And even if it could, maybe it would better wait for FF to fix it?
So it can be posted just as a tip for GH users, but probably applying to the sidebar in general, whenever you use an addon or sth else in it. It may not really fit in the issue pages but I think it makes sense though to be shared, so?...
Someone could say if FF has the same behavior for him, if this post has been useful or not...
Maybe one day I would ask FF about this, that I rather consider as a bug, or a flaw, I think it should remember sidebars presence or not even if leaving from a window with a sidebar in it.

*

(With more details.)

Personnally I always use GH in sidebars instead of the default popup, and I leave them open only in windows where I read several tabs in them, as having the other sidebars closed saves ressources (memory, battery...), esp. for a session with 5000 tabs.
But until now when I closed FF and reopened it, sidebars opened in ALL windows, even if only 2 out of 16 had one before closing.
So of course each time FF took more time to open, and afterwards I also had to close one by one all the useless sidebars. Closing them gives me back ~400MB (useful for my only 4GB RAM).

But I made some tests, and with luck during it I found the solution (even if it is not a fix).
They were a priori finally useless tests, but I can tell.
I restarted with unchecking (not exact words, translated from french) [  ]`Empty history during Firefox closing` in `about:preferences#privacy`, and disabling my few privacy/security addons, in case one of these would have erased the needed information from FF memory, but no difference.

During this, a stupid extra test came in my mind, to close FF before reopening it, but from a window with no sidebar. And oh surprise it worked!
And it still worked after I rechecked the FF setting and re-enabled the addons.

Also, when during a same FF session, GH is disable/enabled, it will open windows with same with/without sidebars even if it is done from a window with a sidebar.
And GH is if disabled before FF closing, re-enabling it in following session will show 0 sidebar.

I checked how much memory I saved by closing my useless sidebars, with System Monitor software, which shows how much memory is consumed (RAM+swap), simply watching the numbers before and after closing, so it was like I said, 400MB.
I also had had the idea that sidebars had charged many informations, and closing them might not suffice to purge them totally, so I did: open `about:memory` in a tab > `[Minimize memory usage]`, which seems to be able to purge this kind of remnants, and it freed 500MB more, so 900MB alltogether (this button is very useful to free memory without losing data). And I did it again but with before: `about:preferences#privacy` > `Remove history` (with everything checked but the cache), but no difference seen.
First I thought the 900MB were all from GH, but later I clicked on the button right after FF opening with GH disabled and it freed 500MB too. So now I count that only 400 come from GH, and I consider that closing the sidebars purges all their informations from FF memory, no remnants (or very few), at least that this button would remove.

**Comment by user0022** [comment]
Hello again!
I can see that this issue is left in open issues page, in a kind of sticky position. I guess it is done because judged important enough to be shown to help users until real conclusion.
So as I opended it, I will try to end it too!

Just for some confirmations:
- After FF ESR (115), I tried also in "classic" FF (126, in Snap version), and it was the same.
- I also tried with another addon in the sidebar (Sidebery), and still the same too.

So I can simply ask about it to Firefox, it is not long to do, but I just have personnally some small things that slow me:
- Personnally I'm a bit slow in general... :/
- My internet access is a bit limited.
- I didn't find yet where to contact FF for this.
 I don't like to make too many accounts on many sites (esp. on ones I dislike like [Github](https://sfconservancy.org/GiveUpGitHub/) or [Firefox](https://searx.be/search?q=firefox%20mitchell%20money%20woke&categories=general&language=en), but they may have good aspects too and I'm no specialist), so I would prefer to stay on Github (now that it's done) for this if possible, but first check showed me 2,4 thousands Mozilla Github "repos" if I remember well. So I have to search a bit more.
 Otherwise never mind I will search for another Mozilla site to sign up, maybe the one for this is [Bugzilla](https://bugzilla.mozilla.org/), or https://ideas.mozilla.org or https://support.mozilla.org or another idk. 

*
Seems I finally found? https://github.com/MozillaFoundation/foundation.mozilla.org
No sure yet, I have to read a bit more, Mozilla has several products, is this where I can ask about Firefox?
I probably will ask here or somewhere else today or before 1 week.

**Comment by madprops** [comment]
The bug tracker is indeed https://bugzilla.mozilla.org , you need a mozilla account.

**Comment by user0022** [comment]
Thank you I didn't have seen this.
In the end my name in Bugzilla is not "user0022", but the ugly name of my email, a bit bothering but well.
I was about to post in the last Github link, oops!!! It doesn't seems to be about Firefox at all, but even reading only titles was hard for my poor english I gave up to know what it is really about.

So I posted it: [Firefox remembers windows with sidebar or not only if window where you leave Firefox has not.](https://bugzilla.mozilla.org/show_bug.cgi?id=1898925).
For my firt post over there I hope I didn't make too many mistakes, there are things here and things there, I apparently can't edit my post.
It will be funny for me if for the first time of my life I have a (very small) influence on an (almost) widespread browser, I will feel like the new Elon Musk.
